internal/transport/transport.go

```diff
@@ -454,26 +459,36 @@ func (s *Stream) write(m recvMsg) {
 	s.buf.put(m)
 }
 
+// Stat returns the streams's underlying *profiling.Stat.
+func (s *Stream) Stat() *profiling.Stat {
+	if s == nil {
```

Comment:
I don't exactly remember, but I vaguely recall facing an issue in some end-to-end tests. I've added a TODO so that someone will re-evaluate this later.